### PR TITLE
F/leaderboard 3

### LINKED ISSUE #3: Feature: User can see a leaderboard of all the high-performing pets
Currently, there are two views, the Home (public) view and the My Pets view (private), we'd like to add a third view to see a leaderboard of all the pets and their current number of points.

## User Story

As a user, I'd like to visit site.com/leaderboard and see a table of the most popular 

## More details

[Wireframe on Miro](https://miro.com/app/board/uXjVO6H5Jjc=/?share_link_id=723904099326)

The table will
- have the following columns: Picture, Name, Points
- render 10 rows (pets) at a time

## Your job

- Using Chakra's [Table component](https://chakra-ui.com/docs/components/table/usage), make a LeaderboardTable.jsx component (in @/components/) that renders 10 pets
- API and Redux: create a reducer, actions and API functions to get the list of pets and store them in Redux
- Route and DB: create a route and database function to get the 10 most popular pets and return them to the client
- Testing!

## Scope

In the future we'd like to add the ability to sort, filter, and paginate, but for now, rendering the 10 most popular pets is fine!

### PR DESCRIPTION
**Pull request by matruane**

### CODE REVIEW

File: client/pages/Leaderboard.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    dispatch(fetchTopTen())
+  }, [])
+
+  return (
+    <>
+      <h3>Pet Points Leaderboard</h3>
+      <PointsTable petScores={petScores} />
+    </>
+  )
+}
```
Comment (joshuavial) [inline client/pages/Leaderboard.jsx:21]: leaderboard component needs a test

File: client/reducers/__tests__/leaderboard.test.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      leaderboard: petsWithScores,
+    }
+    const inputState = { data: [], error: null, loading: false }
+    const expectedOutputState = petsWithScores
+    const actualOutputState = leaderboardReducer(inputState, action)
+
+    expect(actualOutputState.leaderboard).toEqual(expectedOutputState)
+    expect(actualOutputState).not.toBe(inputState)
+  })
+})
```
Comment (joshuavial) [inline client/reducers/__tests__/leaderboard.test.js:44]: setError and requestData parts of reducer don't have a test

**Review by joshuavial — CHANGES_REQUESTED** [review]
Looking good, just a couple of missing tests and you're good to go

File: client/pages/Leaderboard.jsx (hunk 1 above)
Comment (rupert-denton) [reply client/pages/Leaderboard.jsx:21]: Hi JV thanks for the feedback we have added test for this componet

File: client/reducers/__tests__/leaderboard.test.js (hunk 2 above)
Comment (rupert-denton) [reply client/reducers/__tests__/leaderboard.test.js:44]: Hi JV thanks for the feedback we have added test for these parts of the reducer. Thanks!

**Review by joshuavial — APPROVED** [review]
nice work, looking good

**Pull request merged**